### ISSUE TITLE
Crash with NetworkReplicated attribute on derived generic classes

### ISSUE THREAD

**Issue by lifeformed** [FlaxEngine/FlaxEngine#1988]
Use this code in a networked project (I used ArizonaFrameworkSample), and put the Test script on any actor:

```using FlaxEngine;
using FlaxEngine.Networking;

namespace Game
{
    public class Generic<T>: FlaxEngine.Object
    {
        [NetworkReplicated]
        public float Value1;
    }

    public class Concrete : Generic<int>
    {
        [NetworkReplicated]
        public float Value2;
    }

    public class Test : Script
    {
        public override void OnEnable()
        {
            if (NetworkManager.IsServer || NetworkManager.IsHost)
            {
                NetworkReplicator.SpawnObject(new Concrete());
            }
        }
    }
}
```

You should get this error: 
<img width="311" alt="image" src="https://github.com/FlaxEngine/FlaxEngine/assets/1567199/256b2bfa-d15f-44ac-bcff-944759bd4f7f">

**Comment by lifeformed** [comment]
I forgot to mention, the error happens when you run a server/host and then connect a client to it.  The server will crash.

**Comment by mafiesto4** [comment]
```
[ 00:05:18.653 ]: [Error] Unhandled Exception: Could not execute the method because either the method itself or the containing type is not fully instantiated.
[ 00:05:18.742 ]: [Warning] Exception has been thrown. Could not execute the method because either the method itself or the containing type is not fully instantiated.
Stack strace:
   at Game.Generic`1.INetworkSerializable_SerializeNative(IntPtr instancePtr, IntPtr streamPtr)
   at FlaxEngine.Networking.NetworkReplicator.<Internal_InvokeSerializer>g____PInvoke|10_0(IntPtr __typeHandle_native, IntPtr __instance_native, IntPtr __stream_native, Byte __serialize_native)
   at FlaxEngine.Networking.NetworkReplicator.Internal_InvokeSerializer(Type typeHandle, IntPtr instance, IntPtr stream, Boolean serialize) in F:\FlaxEngine\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs:line 41463
   at FlaxEngine.Networking.NetworkReplicator.InvokeSerializer(Type type, Object instance, NetworkStream stream, Boolean serialize) in F:\FlaxEngine\Source\Engine\Networking\NetworkReplicator.cs:line 92
   at Game.Concrete.INetworkSerializable_Serialize(NetworkStream stream)
   at Game.Concrete.INetworkSerializable_SerializeNative(IntPtr instancePtr, IntPtr streamPtr)[ 00:05:18.743 ]: [Warning] Exception has been thrown.
Could not execute the method because either the method itself or the containing type is not fully instantiated.An unhandled exception of type 'System.InvalidOperationException' occurred in FlaxEngine.CSharp.dll
Could not execute the method because either the method itself or the containing type is not fully instantiated.
```

It looks like generated network replication code tries to use generic class without type specified:
```cs
public class Concrete : Generic<int>
{
  [NetworkReplicated]
  public float Value2;

  public new void INetworkSerializable_Serialize(NetworkStream stream)
  {
    NetworkReplicator.InvokeSerializer((Type) Type.GetTypeFromHandle((RuntimeTypeHandle) __typeref (Generic<>)), (FlaxEngine.Object) this, stream, true);
    stream.WriteSingle(this.Value2);
  }
...
```

**Comment by mafiesto4** [comment]
Fixed crash in 4a19657e7524d5ca6524013b0f1c06ef72f9fdde.
Codegen for generic C# classes needs some improvements so I added warning for now: 196a4ffe4927956559e6d0b9c13d511167b5ff5b
